models/DataObject/ClassDefinition/Data/Block.php

```diff
@@ -847,6 +847,7 @@ public function preGetData(mixed $container, array $params = []): mixed
                     $this->markLazyloadedFieldAsLoaded($container);
                 }
             }
+            $this->preSetData($container, $data, $params);
```

Comment:
@dvesh3 @kingjia90 it seems to work, but semantically it somehow feels wrong to call `preSetData` within `preGetData` ... might have some unwanted side-effects in the future, when `preSetData` gets additional functionality and developer is not aware of that is also called during `preGetData` - especially performance wise ...